### ISSUE TITLE
library SONAME went backwards in 5.9.2

### ISSUE THREAD

**Issue by csmall** [net-snmp/net-snmp#432]
The SONAME for libraries is set by the LIBCURRENT and LIBAGE defined in Makefile.top

5.9.1 had a LIBCURRENT of 41 and a LIBAGE of 1 giving a SONAME of 40 (41-1)
https://github.com/net-snmp/net-snmp/blob/v5.9.1/Makefile.top#L85

5.9.2 has a LIBCURRENT of 41 and a LIBAGE of 2 giving a SONAME of 39 (41-2)
https://github.com/net-snmp/net-snmp/blob/v5.9.2/Makefile.top#L85

So only age got incremented which isn't actually a valid change. Age only gets set to zero or increments when Current does too.

So from a C:A:R of 41:1:0 you can go either
* 45:0:0 for removed/changed interfaces
* 42:2:0 for added interface
* 41:1:1 for internal improvements.

My guess is the silly ordering (C:R:A) used in the notes above which comes from the libtool docs got confused with the sensible ordering used in the Makefile (C:A:R) and the R should have been incremented, not the Age.

And yes, libtool docs have bitten me this way too.

**Comment by hardaker** [comment]
Grr....  I *normally* get these right, but yes they can be a pain to figure out and it seems every decade or so you mess them up.  Wish this had been caught during one of the pre-releases rather than now.  This likely warrants a 5.9.3.

**Comment by csmall** [comment]
Yeah I only checked the compiled/linked against OpenSSL v3.0 libraries, I was nearly going to put a rc1 into Debian experimental but ran out of time. It was the Debian linter that snagged it because we have a policy libfoo39.so goes into package libfoo39 not libfoo40 so it was telling me my package name was wrong. Not sure if there is any way of writing a test for this, maybe a minimum SONAME check.

There are only two sorts of people in the world: people who get this wrong, and people who never use libtool.
